lib/events/events.proto

```diff
@@ -937,6 +937,75 @@ message KubeRequest {
     int32 ResponseCode = 11 [ (gogoproto.jsontag) = "response_code" ];
 }
 
+// AppSessionStartEvent is emitted when a user is issued an application certificate.
+message AppSessionStart {
+    // Metadata is a common event metadata
+    Metadata Metadata = 1
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // User is a common user event metadata
+    UserMetadata User = 2
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // SessionMetadata is a common event session metadata
+    SessionMetadata Session = 3
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // ServerMetadata is a common server metadata
+    ServerMetadata Server = 4
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // ConnectionMetadata holds information about the connection
+    ConnectionMetadata Connection = 5
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // PublicAddr is the public address of the application being requested.
+    string PublicAddr = 7 [ (gogoproto.jsontag) = "public_addr" ];
+}
+
+// AppSessionCreateEvent is emitted at the start of a 5 minute chunk on each
+// proxy. This chunk is used to buffer 5 minutes of audit events at a time
+// for applications.
+message AppSessionCreate {
+    // Metadata is a common event metadata
+    Metadata Metadata = 1
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // User is a common user event metadata
+    UserMetadata User = 2
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // SessionMetadata is a common event session metadata
+    SessionMetadata Session = 3
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // ServerMetadata is a common server metadata
+    ServerMetadata Server = 4
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // ConnectionMetadata holds information about the connection
+    ConnectionMetadata Connection = 5
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // SessionChunkID is the ID of the session that was created for this 5 minute
+    // application log chunk.
+    string SessionChunkID = 6 [ (gogoproto.jsontag) = "session_chunk_id" ];
+}
+
+// AppSessionRequestEvent is an HTTP request and response.
+message AppSessionRequest {
+    // Metadata is a common event metadata
+    Metadata Metadata = 1
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
```

Comment:
It can, but I tried to make it a parallel of session bytes, just contains the request/response. Identity information is stored within the start and create events.